### PR TITLE
Bug 798704 - [Email] Clearing the search field by hitting the x icon in ...

### PR DESCRIPTION
**Pull request by rsajdok**
...the search field should reset the search results list

### CODE REVIEW

**Comment by try-server-hook** [comment]
[rsajdok Robert Sajdok (rsajdok)](https://github.com/rsajdok) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e0c621eb69c8)

**Comment by try-server-hook** [comment]
[rsajdok Robert Sajdok (rsajdok)](https://github.com/rsajdok) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1272bbc151dd)

File: apps/email/js/cards/message_list.js [hunk 1]
```diff
@@ -751,6 +751,21 @@ return [
...
+      this.curFolder = model.folder;
+      this.vScroll.clearDisplay();
+
+      headerCursor.startSearch(this.curPhrase, {
+        author: this.curFilter,
+        recipients: this.curFilter,
+        subject: this.curFilter,
+        body: this.curFilter
+      });
+    },
```
Comment (jrburke) [inline apps/email/js/cards/message_list.js]: I think it is just enough to call showSearch with an empty string, that way we can avoid some duplication. So, something just like `this.showSearch('', this.curFilter);` looks like all that is needed.

**Comment by try-server-hook** [comment]
[rsajdok Robert Sajdok (rsajdok)](https://github.com/rsajdok) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=39201a184287)

**Pull request merged**